zulipterminal/ui_tools/views.py

```diff
@@ -1292,8 +1292,10 @@ def __init__(self, controller: Any, stream_id: int) -> None:
             )
         ]
 
-        if "stream_post_policy" in stream:
-            stream_policy = STREAM_POST_POLICY[stream["stream_post_policy"]]
+        if controller.model.get_stream_post_policy(self.stream_id):
+            stream_policy = STREAM_POST_POLICY[
+                controller.model.get_stream_post_policy(self.stream_id)
+            ]
         else:
             if stream.get("is_announcement_only"):
```

Comment:
This existing UI code incorporates implicit checking of the server version, which would be better wrapped into the model under `get_stream_post_policy`, rather than the two functions introduced in this commit and the next.